includes/content-distribution/class-api.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Newspack Network Content Distribution API.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack_Network\Content_Distribution;
+
+use Newspack_Network\Content_Distribution;
+use WP_Error;
+
+/**
+ * API Class.
+ */
+class API {
+	/**
+	 * Initialize hooks.
+	 */
+	public static function init() {
+		add_action( 'rest_api_init', [ __CLASS__, 'register_routes' ] );
+	}
+
+	/**
+	 * Register the REST API routes.
+	 */
+	public static function register_routes() {
+		register_rest_route(
+			'newspack-network/v1',
+			'/content-distribution/distribute/(?P<post_id>\d+)',
+			[
+				'methods'             => 'POST',
+				'callback'            => [ __CLASS__, 'distribute' ],
+				'args'                => [
+					'urls' => [
+						'type'     => 'array',
+						'required' => true,
+						'items'    => [
+							'type' => 'string',
+						],
+					],
+				],
+				'permission_callback' => function() {
+					return current_user_can( 'edit_posts' ); // @TODO Custom capability.
+				},
+			]
+		);
+	}
+
+	/**
+	 * Distribute a post to the network.
+	 *
+	 * @param \WP_REST_Request $request The REST request object.
+	 * @return \WP_REST_Response
+	 */
+	public static function distribute( $request ) {
+		$post_id = $request->get_param( 'post_id' );
+		$urls    = $request->get_param( 'urls' );
+
+		try {
+			$outgoing_post = new Outgoing_Post( $post_id );
+		} catch ( \InvalidArgumentException $e ) {
+			return new WP_Error( 'newspack_network_content_distribution_error', $e->getMessage(), [ 'status' => 400 ] );
```

Comment:
We should also wrap the errors in rest_ensure_response()

Reply:
`WP_Error` objects are handled differently by the rest API controller. This function will not process it: https://github.com/WordPress/wordpress-develop/blob/e9dfa8c34c74cb51da93ae4b4e61c1df0378b3d7/src/wp-includes/rest-api.php#L659-L662

Reply:
Can we fix the return type then please? My fancy editor is freaking out ;)

Reply:
Ah I see! Absolutely, I'd never beef with phpstorm

Reply:
35c1b46a094b96ffde5ccf61fbdf46ac4d089505